### DISCUSSION TITLE
What is the difference between login and non-login shell in msys2?

### DISCUSSION THREAD

**Discussion by mon-jai** [msys2/MSYS2-packages#2882] [Q&A] [answered]
With `msys2_shell.cmd`, it takes about 3-4 seconds to stratup msys2 bash.

```
"C:/msys64/msys2_shell.cmd" -defterm -here -no-start -msys
```

But without the login flag, bash starts up almost immediately.

```
cmd /k set MSYSTEM=MSYS&& set PATH=C:/msys64/usr/bin;%PATH%&& bash
```

So what's the difference and suitable use cases of the two startup methods?

**Comment by Biswa96** [comment, marked as answer]
In case of login shell, commands are executed from some files. More details can be found here https://www.gnu.org/software/bash/manual/html_node/Bash-Startup-Files.html. Also you can find some information in bash man page https://man7.org/linux/man-pages/man1/bash.1.html

**Reply by mon-jai** [reply]
So there are only differences on the upsteam (bash) side, but not msys2 side?

**Reply by lazka** [reply]
It will execute /etc/profile for example. which is provided by msys2
